### ISSUE TITLE
cljol's view doesn't play well with large data structures

### ISSUE THREAD

**Issue by jumarko** [jafingerhut/cljol#2]
Trying this on any vector containing more than a few hundreds of elements is painful.
Not only the rendering process takes very long but the resulting picture is unusable.
I tried it with `(cljol.dig9/view (vec (range 1000)))`.

It would be nice if the visualizations were limited to certain "length" / "depth" and parent nodes also contained a "summary" size for all their children together.

**Comment by jafingerhut** [comment]
Agreed that large dumps of data are too much for a person to deal with :)

A summary size for all objects reachable from a given object is a nice addition to what is there now, as long as you realize that those can "overlap", i.e. a common set of objects might be reachable both from objects A and B, although neither A nor B can get to each other by following chains of references.

A depth restriction on what is displayed - yes, also sounds useful.

I am not sure what you mean by a length limit?  As in, maximum number of references out of one object that are included?  Which subset would you pick if the limit was exceeded?

**Comment by jumarko** [comment]
I don't have a super-clear idea how this should work but I was thinking about something similar to [*print-length*](https://clojuredocs.org/clojure.core/*print-length*)

**Comment by jafingerhut** [comment]
If you try the latest version it uses the ubergraph library to create an intermediate graph before drawing, so all of loom's and ubergraph's algorithms are available for examining it and creating other graphs.

I have implemented the "total size" of all reachable objects, as well as the number of reachable objects, as a label for every object in the graph.  Also a function that gives some summary statistics about the graph in text form.

It does not yet implement the depth limit, but that will be a pretty small amount of work from its current state.

**Comment by jumarko** [comment]
That's cool, thanks!

**Comment by jafingerhut** [comment]
OK, code and README on master branch updated a few minutes ago, with examples showing how to create the full graph for one or more objects, then to trim it down in a couple of ways: remove all leaf nodes (which could be repeated), or to limit the node's distance in reference hops away from one of the starting objects.

Oh, and also a function that prints summary statistics about the graph, without showing it, so you can decide whether you want to show it, or first use one of the trimming functions.
